services/background-worker/src/Wire/BackendNotificationPusher.hs

```diff
@@ -38,13 +43,19 @@ instance RabbitMQEnvelope Q.Envelope where
   ack = Q.ackEnv
   reject = Q.rejectEnv
 
-pushNotification :: RabbitMQEnvelope e => Domain -> (Q.Message, e) -> AppT IO ()
-pushNotification targetDomain (msg, envelope) = do
-  -- Jittered exponential backoff with 10ms as starting delay and 300s as max
-  -- delay. When 300s is reached, every retry will happen after 300s.
+pushNotification :: RabbitMQEnvelope e => MVar () -> Domain -> (Q.Message, e) -> AppT IO ()
+pushNotification mvar targetDomain (msg, envelope) = do
+  -- Jittered exponential backoff with 10ms as starting delay and 20s as maximum
+  -- cumulative delay. When 20s is cumulatively reached, the request will fail.
   --
-  -- FUTUREWORK: Pull these numbers into config
-  let policy = capDelay 300_000_000 $ fullJitterBackoff 10000
+  -- FUTUREWORK: Pull these numbers into config.
+  -- FUTUREWORK: Is this futurework still needed given the new changes?
+  -- Limit retries to a max of 20 seconds, as this is what Kubernetes is going
+  -- to give us for a SIGTERM shutdown notice by default. If we take longer than
+  -- that, it will SIGKILL the pod and there is nothing we can do to stop that.
+  -- If we fail after this, the notification will be NACKed, and will be redelivered
+  -- by RabbitMQ for another attempt.
+  let policy = limitRetriesByCumulativeDelay (20 * 1_000_000) $ fullJitterBackoff 10000
```

Comment:
So we still need a config update. Setting the max delay to the kubernetes grace timeout isn't going to work because that won't factor in processing time. We will need a fudge factor if we are setting those values to be equal.